### ISSUE TITLE
Error in AZ-203_02_lab Exercise 3

### ISSUE THREAD

**Issue by multco-malan** [MicrosoftLearning/AZ-203-DevelopingSolutionsforMicrosoftAzure#17]
# Module: AZ-203-DevelopingSolutionsforMicrosoftAzure
#### Step: 10

Description of issue
Thumbnail images are not generated.
So I went to the Test tab in the azure portal for the function.
Entered an image i'd already uploaded.
https://self.blob.core.windows.net/images/grilledcheese.jpg as the request body.

The output window shows the error:

2019-05-28T20:07:59.055 [Information] Executing 'Functions.ImageManagerBlobTrigger1' (Reason='This function was programmatically called via the host APIs.', Id=cf97f4b1-c985-4fd7-ad17-acba2f7d636a)
2019-05-28T20:07:59.539 [Error] Function compilation error
Microsoft.CodeAnalysis.Scripting.CompilationErrorException : Script compilation failed.
   at async Microsoft.Azure.WebJobs.Script.Description.DotNetFunctionInvoker.CreateFunctionTarget(CancellationToken cancellationToken) at C:\azure-webjobs-sdk-script\src\WebJobs.Script\Description\DotNet\DotNetFunctionInvoker.cs : 314
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.Azure.WebJobs.Script.Description.FunctionLoader`1.GetFunctionTargetAsync[T](Int32 attemptCount) at C:\azure-webjobs-sdk-script\src\WebJobs.Script\Description\FunctionLoader.cs : 55
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.Azure.WebJobs.Script.Description.DotNetFunctionInvoker.GetFunctionTargetAsync(Boolean isInvocation) at C:\azure-webjobs-sdk-script\src\WebJobs.Script\Description\DotNet\DotNetFunctionInvoker.cs : 183
2019-05-28T20:07:59.861 [Error] run.csx(1,7): error CS0246: The type or namespace name 'SixLabors' could not be found (are you missing a using directive or an assembly reference?)
2019-05-28T20:08:00.200 [Error] run.csx(2,7): error CS0246: The type or namespace name 'SixLabors' could not be found (are you missing a using directive or an assembly reference?)

**Comment by seesharprun** [comment]
This indicates to me that you may have missed the step to upload the **function.proj** file. This file includes the NuGet package reference necessary to import the **SixLabors.ImageSharp** package:

```
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <TargetFramework>netstandard2.0</TargetFramework>
    </PropertyGroup>
    <ItemGroup>
        <PackageReference Include="SixLabors.ImageSharp" Version="1.0.0-beta0006" />
    </ItemGroup>
</Project>
```

Please review **Exercise 3**, **Task 2**, **Step 8** again.

I also added a note to the lab instructions to explain what this is doing for future MCTs. This note is available in [Version 20190701.09](https://github.com/MicrosoftLearning/AZ-203-DevelopingSolutionsforMicrosoftAzure/releases/tag/v20190701.09).

**Comment by molenaar** [comment]
Problem is back again. No thumbnails at all. I did this labs many times and it worked before. This week it doesn't. It doesn't work for T02 and for T04 either. Very frustrating for students, they do the lab over and over again, and it still doesn't work. 
Downloading packages fails and after that depencency are missing.  The latest function.proj file is uploaded.

**Comment by molenaar** [comment]
2019-07-17T13:48:47.451 [Information] Restoring packages.
2019-07-17T13:48:47.483 [Information] Starting packages restore
2019-07-17T13:48:48.292 [Information] System.UnauthorizedAccessException: Attempted to perform an unauthorized operation.
2019-07-17T13:48:48.293 [Information] at Microsoft.Win32.RegistryKey.Win32Error(Int32 errorCode, String str)
2019-07-17T13:48:48.293 [Information] at Microsoft.Win32.RegistryKey.SetValue(String name, Object value, RegistryValueKind valueKind)
2019-07-17T13:48:48.293 [Information] at System.Environment.SetEnvironmentVariableCore(String variable, String value, EnvironmentVariableTarget target)
2019-07-17T13:48:48.293 [Information] at System.Environment.SetEnvironmentVariable(String variable, String value, EnvironmentVariableTarget target)
2019-07-17T13:48:48.293 [Information] at Internal.Runtime.Augments.EnvironmentAugments.SetEnvironmentVariable(String variable, String value, EnvironmentVariableTarget target)
2019-07-17T13:48:48.294 [Information] at System.Environment.SetEnvironmentVariable(String variable, String value, EnvironmentVariableTarget target)
2019-07-17T13:48:48.294 [Information] at Microsoft.DotNet.Cli.Utils.EnvironmentProvider.SetEnvironmentVariable(String variable, String value, EnvironmentVariableTarget target)
2019-07-17T13:48:48.294 [Information] at Microsoft.DotNet.ShellShim.WindowsEnvironmentPath.AddPackageExecutablePathToUserPath()
2019-07-17T13:48:48.294 [Information] at Microsoft.DotNet.Configurer.DotnetFirstTimeUseConfigurer.Configure()
2019-07-17T13:48:48.294 [Information] at Microsoft.DotNet.Cli.Program.ConfigureDotNetForFirstTimeUse(INuGetCacheSentinel nugetCacheSentinel, IFirstTimeUseNoticeSentinel firstTimeUseNoticeSentinel, IAspNetCertificateSentinel aspNetCertificateSentinel, IFileSentinel toolPathSentinel, Boolean hasSuperUserAccess, DotnetFirstRunConfiguration dotnetFirstRunConfiguration, IEnvironmentProvider environmentProvider)
2019-07-17T13:48:48.294 [Information] at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, ITelemetry telemetryClient)
2019-07-17T13:48:48.295 [Information] at Microsoft.DotNet.Cli.Program.Main(String[] args)
2019-07-17T13:48:48.305 [Information] Packages restored.
2019-07-17T13:48:48.608 [Information] Script for function 'ImageManager' changed. Reloading.
2019-07-17T13:48:48.841 [Error] run.csx(1,7): error CS0246: The type or namespace name 'SixLabors' could not be found (are you missing a using directive or an assembly reference?)
2019-07-17T13:48:48.956 [Error] run.csx(2,7): error CS0246: The type or namespace name 'SixLabors' could not be found (are you missing a using directive or an assembly reference?)
2019-07-17T13:48:49.021 [Error] run.csx(3,7): error CS0246: The type or namespace name 'SixLabors' could not be found (are you missing a using directive or an assembly reference?)
2019-07-17T13:48:49.088 [Error] run.csx(4,7): error CS0246: The type or namespace name 'SixLabors' could not be found (are you missing a using directive or an assembly reference?)
2019-07-17T13:48:49.187 [Error] run.csx(5,7): error CS0246: The type or namespace name 'SixLabors' could not be found (are you missing a using directive or an assembly reference?)
2019-07-17T13:48:49.241 [Error] run.csx(10,12): error CS0246: The type or namespace name 'Image<>' could not be found (are you missing a using directive or an assembly reference?)
2019-07-17T13:48:49.317 [Error] run.csx(10,18): error CS0246: The type or namespace name 'Rgba32' could not be found (are you missing a using directive or an assembly reference?)
2019-07-17T13:48:49.391 [Error] run.csx(10,34): error CS0103: The name 'Image' does not exist in the current context
2019-07-17T13:48:49.448 [Error] run.csx(15,36): error CS0246: The type or namespace name 'JpegEncoder' could not be found (are you missing a using directive or an assembly reference?)
2019-07-17T13:48:49.509 [Information] Compilation failed.

**Comment by seesharprun** [comment]
It looks like the functions service failed to download the packages. These two lines lead me to believe that is the issue:

```
2019-07-17T13:48:47.483 [Information] Starting packages restore
2019-07-17T13:48:48.292 [Information] System.UnauthorizedAccessException: Attempted to perform an unauthorized operation.
```

I am going to look at the lab and see if I can replicate this issue.

You mentioned that you got a similar error in T04. Can you create a new issue for T04 so we can track that one too?

**Comment by seesharprun** [comment]
I just did a quick test and I am also unable to reference the library imported from NuGet.

**Comment by seesharprun** [comment]
I don't want you to think I forgot this issue. I simply haven't been able to get this to work successfully.

I am now going to try and get this product team involved since this effectively blocks two labs.

**Comment by seesharprun** [comment]
@ToshikazuOki , I have confirmed that adding that application setting does fix the issue. I still would like the product team to chime in but I will rewrite labs 2 & 4 this week to resolve this issue and issue #49.

**Comment by seesharprun** [comment]
This bug was resolved in [Version 20190725.03](https://github.com/MicrosoftLearning/AZ-203-DevelopingSolutionsforMicrosoftAzure/releases/tag/v20190725.03).
